modules/measurement/budget-calculator/load.php

```diff
@@ -0,0 +1,165 @@
+}
+
+/**
+ * Register the option page.
+ *
+ * @since 1.0.0
+ */
+function budget_calc_menu_page() {
+	add_menu_page(
+		'Budget Calculator',
+		'Budget Calculator',
+		'manage_options',
+		'budget-calculator',
+		'budget_calc_page_content',
+		'dashicons-calculator'
+	);
+}
+add_action( 'admin_menu', 'budget_calc_menu_page' );
+
+/**
+ * Register the settings for the option page.
+ *
+ * @since 1.0.0
+ */
+function budget_calc_register_settings() {
+	register_setting( 'budget_calc_settings', 'budget_calc_options' );
+	add_settings_section( 'budget_calc_settings_id', 'Budget Calculator Metrics', '', 'budget-calculator' );
+
+	$options        = get_option( 'budget_calc_options', array() );
+	$range_settings = array(
+		'HTML'       => 300,
+		'CSS'        => 200,
+		'Font'       => 200,
+		'Images'     => 3000,
+		'JavaScript' => 750,
+	);
+
+	foreach ( $range_settings as $title => $max ) {
+		$id = strtolower( $title ) . '_range';
+
+		add_settings_field(
+			$id,
+			$title,
+			function () use ( $options, $id, $max ) {
+				budget_calc_render_range_field( $options, $id, $max );
+			},
+			'budget-calculator',
+			'budget_calc_settings_id'
+		);
+	}
+}
+add_action( 'admin_init', 'budget_calc_register_settings' );
+
+/**
+ * Render a range input field for the budget calculator options page.
+ *
+ * @since 1.0.0
+ *
+ * @param array  $options  The settings value already set in the database, if any.
+ * @param string $field_id The ID of the field in the database.
+ * @param int    $max      The maximum value that can be set on the range input.
+ */
+function budget_calc_render_range_field( $options, $field_id, $max ) {
+	$text     = isset( $options[ $field_id ] ) ? esc_attr( $options[ $field_id ] ) : 0;
+	$field_id = esc_attr( $field_id );
+	$max      = esc_attr( $max );
+
+	echo "<input type='range' id='$field_id' name='budget_calc_options[$field_id]' min='0' max='$max' step='10'
+		value='$text' oninput='document.getElementById(\"${field_id}_output\").value = this.value'/>
+		<output id='${field_id}_output' for='budget_calc_options[$field_id]'>${text}</output><span>KB</span>";
+}
+
+/**
+ * Render the budget.json file after saving the options page.
+ *
+ * @since 1.0.0
+ *
+ * @param array $value The newly set value for this option.
+ */
+function budget_calc_render_json( $value ) {
+	$resource_sizes = array();
+	$type_mapping   = array(
+		'html_range'       => 'document',
+		'css_range'        => 'stylesheet',
+		'font_range'       => 'font',
+		'images_range'     => 'image',
+		'javascript_range' => 'script',
+	);
+
+	foreach ( $type_mapping as $range => $type ) {
+		if ( isset( $value[ $range ] ) ) {
+			$resource_sizes[] = array(
+				'resourceType' => $type,
+				'budget'       => $value[ $range ],
+			);
+		}
+	}
+
+	$data = array( array( 'resourceSizes' => $resource_sizes ) );
+
+	header( 'Content-disposition: attachment; filename=budget.json' );
+	header( 'Content-type: application/json' );
+	echo wp_json_encode( $data );
+	die;
```

Comment:
Replace with `wp_die` instead.

Reply:
I've followed your other suggestion of using `wp_send_json` and that calls `wp_die` already, so no need for a call to `die` there anymore!